File: src/main/java/com/tribhaskar/springkafkago/model/Notification.java
package com.tribhaskar.springkafkago.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    private String id;
    private String recipient;
    private String subject;
    private String content;
    private String type; // SMS, EMAIL, PUSH, etc.
    private LocalDateTime timestamp;
    private boolean read;

    // Additional constructor for convenience
    public Notification(String recipient, String subject, String content, String type) {
        this.recipient = recipient;
        this.subject = subject;
        this.content = content;
        this.type = type;
        this.timestamp = LocalDateTime.now();
        this.read = false;
    }
}

File: go-kafka-integration/models/notification.go
package models

import (
	"time"
)

// Notification represents a notification message in the system
type Notification struct {
	ID        string    `json:"id"`
	Recipient string    `json:"recipient"`
	Subject   string    `json:"subject"`
	Content   string    `json:"content"`
	Type      string    `json:"type"` // EMAIL, SMS, PUSH
	Timestamp time.Time `json:"timestamp"`
	Read      bool      `json:"read"`
}

// NotificationType enum
const (
	EMAIL = "EMAIL"
	SMS   = "SMS"
	PUSH  = "PUSH"
)